### ISSUE TITLE
Feature Request: Get / Set schedule

### ISSUE THREAD

**Issue by steelixb** [gndean/home-assistant-hypervolt-charger#24]
It would be super useful if we could get and set the schedule. I don't understand the limitation that prevents this right now, as I have various other integrations that allow me to get and set dates and times. As an example, I change the scheduled charging of my house battery through a service, passing a JSON payload:

```yaml
service: givenergy_local.enable_timed_charge
data: >
  {% set startTime = now() %}
  {% set endTime = as_datetime(states('sensor.octopus_intelligent_offpeak_end')).astimezone() %}
  {
    "start_time": "{{ '%02d' % startTime.hour }}:00:00",
    "end_time": "{{ '%02d' % endTime.hour }}:{{ '%02d' % endTime.minute }}:00",
    "device_id": "xxxx"
  }
```

This would be useful for charging one of our cars that doesn't support Intelligent Octopus or specifying a target battery SoC, as we could automate scheduling to always charge to 80%

**Comment by gndean** [comment]
Thanks for the suggestion. I'd ideally like something more user friendly than having to pass JSON. That's fine for the most tech savvy users, but not for everyone.
Do you have any examples of other integrations that allow getting and setting of dates and times (or ideally schedules) in other ways?

**Comment by steelixb** [comment]
It looks like the Intelligent Octopus integration allows you to set a Target Ready By Time which may be similar?
![image](https://github.com/gndean/home-assistant-hypervolt-charger/assets/23078066/71a1a06b-7df1-42e4-bab3-81c9895ea3bc)

**Comment by gndean** [comment]
> Could this also be relevant?
> https://www.home-assistant.io/integrations/datetime/

> Since version 2023.6

Yes, thanks, it is! Now, to decide on how to set a variable numbers of start / stop times. Maybe the integration could just define a fixed number of start and stop pairs of times (4, say). And any set where start=stop would be ignored. So someone could just set the `start1` and `end1` fields to set a single on/off schedule.

Or something like that?

**Comment by steelixb** [comment]
I think it would be best if initially this just reads / writes the Hypervolt's schedule. That way you can still use the app to modify it. To support multiple schedules, we could have an automation blueprint to handle it

**Comment by gndean** [comment]
Agreed, and that's what I mean too. The integration would read/write the same schedule as used by the HV app. But the HA integration needs some way to represent the schedule within HA, and the schedule consists of an arbitrary number of start/stop times.

**Comment by steelixb** [comment]
Ahh, yes I totally forgot you could set multiple times. Yeah, I think a fixed number of start / stop pairs makes sense.

**Comment by gndean** [comment]
I've released V1.3.0 which has support for viewing and setting the schedule session start and end times. See the Readme for details. It uses the Home Assistant time format, supported from 2023.6

Note that Hypervolt appear to have had some issues with their APIs today - changing session times within the Hypervolt app and this integration sporadically failed. So you may see an occasional error when trying to apply a new schedule but unless you see a consistent error, this is likely to just be a (hopefully) transient issue with the Hypervolt APIs.

Let me know how you get on with it.
